homeassistant/components/motionblinds_ble/sensor.py

```diff
@@ -0,0 +1,221 @@
+"""Sensor entities for the Motionblinds BLE integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from math import ceil
+
+from motionblindsble.const import (
+    MotionBlindType,
+    MotionCalibrationType,
+    MotionConnectionType,
+)
+from motionblindsble.device import MotionDevice
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    ATTR_BATTERY,
+    ATTR_CALIBRATION,
+    ATTR_CONNECTION,
+    ATTR_SIGNAL_STRENGTH,
+    CONF_MAC_CODE,
+    DOMAIN,
+    ICON_CALIBRATION,
+    ICON_CONNECTION,
+)
+from .entity import MotionblindsBLEEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+@dataclass(frozen=True, kw_only=True)
+class MotionblindsBLESensorEntityDescription(SensorEntityDescription):
+    """Entity description of a sensor entity with native_value attribute."""
+
+    native_value: str | None = None
+
+
+SENSOR_TYPES: dict[str, MotionblindsBLESensorEntityDescription] = {
+    ATTR_BATTERY: MotionblindsBLESensorEntityDescription(
+        key=ATTR_BATTERY,
+        translation_key=ATTR_BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    ATTR_CONNECTION: MotionblindsBLESensorEntityDescription(
+        key=ATTR_CONNECTION,
+        translation_key=ATTR_CONNECTION,
+        icon=ICON_CONNECTION,
+        device_class=SensorDeviceClass.ENUM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        options=[connection_type.value for connection_type in MotionConnectionType],
```

Comment:
Please don't use an enum from the lib as it could break in the future.

Reply:
@edenhaus related to the options enum:

Is the `options` key necessary? Here I added it for the connection sensor, but not for the calibration sensor. Also I added a key `initial_value` here which sets the initial value of the sensor, can this be the enum value or should I also use a string here?

Reply:
Options key is necessary because it enhances the user experience with automating.

You can instead pass in a list of strings for example.

Reply:
Fixed it, I now use str lists for both options